low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -357,3 +381,190 @@ def get_nw_status(self, nw_interface, interface):
             nw_cable_conn_status = "UNKNOWN"
             # Log the error when logging class is in place.
         return nw_status, nw_cable_conn_status
+
+    def initialize_systemd(self):
+        """Initialize system bus and interface."""
+        self._bus = SystemBus()
+        self._systemd = self._bus.get_object(
+            SYSTEMD_BUS, "/org/freedesktop/systemd1")
+        self._manager = Interface(self._systemd, dbus_interface=MANAGER_IFACE)
+
+    @staticmethod
+    def initialize_cortx_build_info():
+        """Initialize Cortx build info."""
+        if os.path.isfile(CORTX_RELEASE_FACTORY_INFO):
+            cortx_build_info_file_path = "%s://%s" % (
+                CONFIG_SPEC_TYPE, CORTX_RELEASE_FACTORY_INFO)
+            Conf.load("cortx_build_info", cortx_build_info_file_path)
+
+    @staticmethod
+    def get_cortx_service_list():
+        """Get list of cortx services for health generation."""
+        # TODO use solution supplied from HA for getting
+        # list of cortx services or by parsing resource xml from PCS
+        cortx_services = [
+            "motr-free-space-monitor.service",
+            "s3authserver.service",
+            "s3backgroundproducer.service",
+            "s3backgroundconsumer.service",
+            "hare-hax.service",
+            "haproxy.service",
+            "sspl-ll.service",
+            "kibana.service",
+            "csm_agent.service",
+            "csm_web.service",
+            "event_analyzer.service",
+        ]
+        return cortx_services
+
+    def get_external_service_list(self):
+        """Get list of external services for health generation."""
+        # TODO use solution supplied from RE for getting
+        # list of external services.
+        external_services = set(Conf.get(
+            SSPL_CONF, f"{self.SERVICE_HANDLER}>{self.SERVICE_LIST}", []))
+        return external_services
+
+    def get_cortx_service_info(self):
+        """Get cortx service info in required format."""
+        service_info = []
+        cortx_services = self.get_cortx_service_list()
+        for service in cortx_services:
+            response = self.get_systemd_service_info(service)
+            if response is not None:
```

Comment:
even if response is None, can't we add the service_name with null response ?

Reply:
Yes, it was discussed in the morning. Earlier the function was returning `None` if service is not installed, now it returns the schema with installed field as `False` and other fields as `NA`

Reply:
good